staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec.go

```diff
@@ -225,6 +225,9 @@ func (r *roundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
 		return nil, fmt.Errorf("getting credentials: %v", err)
 	}
 	if creds.token != "" {
+		if req.Header == nil {
```

Comment:
All the constructors I can find already do this, if this is necessary it must have been constructed wrong?

Reply:
e.g. looking at https://golang.org/pkg/net/http/#Request

Reply:
I looked at your other PR, I prefer that fix--people calling this need to construct the thing correctly. I'm not sure that there's not some other problem with the construction.

Reply:
Fair enough. Let's consider cases where people don't use a constructor a mistake in which case they deserve a panic :)